mp34dt01_l4_Driver.c

```diff
@@ -0,0 +1,904 @@
+static uint8_t DFSDMx_DeInit(void);
+static void DFSDMx_ChannelMspDeInit(DFSDM_Channel_HandleTypeDef *hdfsdm_channel);
+static void DFSDMx_FilterMspDeInit(DFSDM_Filter_HandleTypeDef *hdfsdm_filter);
+
+
+
+
+/**
+* @}
+*/
+
+/** @defgroup MP34DT01_L4_DRIVER_AUDIO_L4_Exported_Functions
+* @{
+*/
+
+/**
+* @brief  Initializes audio acquisition.
+* @param  AudioFreq: Audio frequency to be configured for the peripherals.
+* 		  Possible values are 8000, 16000, 32000, 48000 OR 96000 Hz
+* @param  BitRes: Not used in this release.
+* @param  ChnlNbr: Number of channel to be configured.
+* @retval AUDIO_OK if correct communication, else wrong communication
+*/
+uint8_t MP34DT01_Init(uint32_t AudioFreq, uint32_t BitRes, uint32_t ChnlNbr)
+{
+  UNUSED(BitRes);
+  uint32_t ret = AUDIO_ERROR;
+
+  /*Set Structure for internal state*/
+  MP34DT01_Handler.MicChannels = ChnlNbr;
+  MP34DT01_Handler.Sampling_Freq = AudioFreq;
+
+  MP34DT01_ClockConfig(MP34DT01_Handler.Sampling_Freq, NULL);
+
+  /* SAI data transfer preparation:
+  Prepare the Media to be used for the audio transfer from memory to SAI peripheral */
+  ret = DFSDMx_Init(AudioFreq, ChnlNbr);
+
+  /* Return AUDIO_OK when all operations are correctly done */
+  return ret;
+}
+
+/**
+* @brief  DeInitializes the audio peripheral.
+* @retval None
+*/
+uint8_t MP34DT01_DeInit(void)
+{
+  return DFSDMx_DeInit();
+}
+
+/**
+* @brief  Clock Config.
+* @param  Params: additional parameters where required
+* @param  AudioFreq: Audio frequency used to play the audio stream.
+* @note   This API is called by MP34DT01_Init()
+*         Being __weak it can be overwritten by the application
+* @retval AUDIO_OK if no problem during execution, AUDIO_ERROR otherwise
+*/
+__weak uint8_t MP34DT01_ClockConfig(uint32_t AudioFreq, void *Params)
+{
+  UNUSED(Params);
+  RCC_PeriphCLKInitTypeDef RCC_ExCLKInitStruct;
+
+  switch (AudioFreq) {
+  case AUDIO_FREQUENCY_8K: {
+    RCC_ExCLKInitStruct.PLLSAI1.PLLSAI1N = 37;
+    RCC_ExCLKInitStruct.PLLSAI1.PLLSAI1P = 17;
+    break;
+  }
+  case AUDIO_FREQUENCY_16K: {
+    RCC_ExCLKInitStruct.PLLSAI1.PLLSAI1N = 37;
+    RCC_ExCLKInitStruct.PLLSAI1.PLLSAI1P = 17;
+    break;
+  }
+  case AUDIO_FREQUENCY_32K: {
+    RCC_ExCLKInitStruct.PLLSAI1.PLLSAI1N = 43;
+    RCC_ExCLKInitStruct.PLLSAI1.PLLSAI1P = 7;
+    break;
+  }
+  case AUDIO_FREQUENCY_48K: {
+    RCC_ExCLKInitStruct.PLLSAI1.PLLSAI1N = 43;
+    RCC_ExCLKInitStruct.PLLSAI1.PLLSAI1P = 7;
+    break;
+  }
+  case AUDIO_FREQUENCY_96K: {
+    RCC_ExCLKInitStruct.PLLSAI1.PLLSAI1N = 43;
+    RCC_ExCLKInitStruct.PLLSAI1.PLLSAI1P = 7;
+    break;
+  }
+  default: {
+    RCC_ExCLKInitStruct.PLLSAI1.PLLSAI1N = 43;
+    RCC_ExCLKInitStruct.PLLSAI1.PLLSAI1P = 7;
+    break;
+  }
+  }
+
+  /* Configure PLLSAI prescalers */
+  /* Please note that some of these parameters must be consistent with
+  the parameters of the main PLL */
+  RCC_ExCLKInitStruct.PeriphClockSelection = RCC_PERIPHCLK_SAI1;

```

Comment:
IS it valid for all series ?
Seems only avaible for F7/F4 and L4....

Reply:
This library is only available for board with DFSDM module => F4/F7/L4 series. But not tested on F4/F7!